caas/kubernetes/provider/specs/types.go

```diff
@@ -72,6 +72,9 @@ type K8sContainerSpec struct {
 	LivenessProbe   *core.Probe           `json:"livenessProbe,omitempty" yaml:"livenessProbe,omitempty"`
 	ReadinessProbe  *core.Probe           `json:"readinessProbe,omitempty" yaml:"readinessProbe,omitempty"`
 	SecurityContext *core.SecurityContext `json:"securityContext,omitempty" yaml:"securityContext,omitempty"`
+
+	EnvFrom []core.EnvFromSource `json:"envFrom,omitempty" yaml:"envFrom,omitempty"`
+	Env     []core.EnvVar        `json:"env,omitempty" yaml:"env,omitempty"`
```

Comment:
We need a slightly different abstraction here.
We already support the "config" keyword for straight pod env key values.

ie
```
    config:
      attr: foo=bar; name["fred"]="blogs";
      foo: bar
```

We need to look to extend that to add in the extra capability, eg

```
    config:
      foo: bar
      my_node-name:
        field:
          name: spec.nodeName
      thing:
         secret:
           name: foo
           key: bar
      thing2:
         secret:
           name: foo
      thing3:
         configMap:
           name: foo
      thing4:
         configMap:
           name: foo
           value: bar
```

So "config" attributes are extended to accept either a string as we have now for simple scalar values, or a map which is used to define the 3 new cases.
There are 3 new keywords: field, configMap, secret.
If only the name is specified for configMap or secret, that maps to EnvFrom where all fields are used. Otherwise we just use the single named field.